src/rust/lib/prelude/src/macros.rs

```diff
@@ -79,3 +79,50 @@ macro_rules! alias {
         impl<T:$($tok)*> $name for T {}
     )*};
 }
+
+
+
+// ==============
+// === Lambda ===
+// ==============
+
+/// Clones all arguments from the first argument list by using `CloneRef` and defines lambda with
+/// arguments from the second argument list (if present). For example, the following usage
+///
+/// ```compile_fail
+/// f! { (a,b)(c) a + b + c }
+/// ```
+///
+/// is equivalent to:
+///
+/// ```compile_fail
+/// {
+///     let a = a.clone_ref();
+///     let b = b.clone_ref();
+///     move |c| { a + b + c }
+/// }
+/// ```
+#[macro_export]
+macro_rules! f {
```

Comment:
God, please, NOOO! Don't make macros with one-letter names :open_mouth:

Reply:
As agreed via phone, this is such an useful function which will be so widely used as something similar to `fn`, that actually name `f!` is not so bad :)